src/app/components/map/map-main/map-main.component.ts

```diff
@@ -8,55 +8,188 @@ Carnegie Mellon(R) and CERT(R) are registered in the U.S. Patent and Trademark O
 DM20-0181
 */
 
-import { Component, OnInit } from '@angular/core';
-import { ActivatedRoute, Router } from '@angular/router';
+import {
+  AfterViewChecked,
+  ChangeDetectorRef,
+  Component,
+  OnInit,
+  TemplateRef,
+  ViewChild,
+} from '@angular/core';
+import { MatDialog, MatDialogRef } from '@angular/material/dialog';
+import { ActivatedRoute } from '@angular/router';
 import { VmMap, VmsService } from '../../../generated/vm-api';
+import { MapTeamDisplayComponent } from '../map-team-display/map-team-display.component';
+import { MapComponent } from '../map.component';
+import { NewMapComponent } from '../new-map/new-map.component';
 
 @Component({
   selector: 'app-map-main',
   templateUrl: './map-main.component.html',
-  styleUrls: ['./map-main.component.css']
+  styleUrls: ['./map-main.component.css'],
 })
-
-export class MapMainComponent implements OnInit {
+export class MapMainComponent implements OnInit, AfterViewChecked {
   VmMaps: VmMap[];
-  selected;
+  selected: VmMap;
   editMode: boolean;
+  readMap: boolean;
+  mapInitialized: boolean;
+  teamId: string;
+  viewId: string;
+  build: boolean;
+  mapId: string;
+  editProps: boolean;
+  creatingMap: boolean;
+
+  @ViewChild(MapTeamDisplayComponent) displayChild: MapTeamDisplayComponent;
+  @ViewChild(MapComponent) buildChild: MapComponent;
+  @ViewChild(NewMapComponent) propertiesDialog: NewMapComponent;
+
+  @ViewChild('newMapDialog') newMapDialog: TemplateRef<NewMapComponent>;
+  @ViewChild('editPropsDialog') editPropsDialog: TemplateRef<NewMapComponent>;
+  private dialogRef: MatDialogRef<NewMapComponent>;
 
   constructor(
     private vmsService: VmsService,
-    private router: Router,
-    private route: ActivatedRoute
-  ) { }
+    private route: ActivatedRoute,
+    private changeDetector: ChangeDetectorRef,
+    private dialog: MatDialog
+  ) {
+    this.mapInitialized = false;
+  }
 
   ngOnInit(): void {
+    this.VmMaps = new Array<VmMap>();
+    this.route.params.subscribe((params) => {
+      this.viewId = params['viewId'];
+    });
     this.getMaps();
-    console.log(this.VmMaps);
+  }
+
+  ngAfterViewChecked() {
+    this.changeDetector.detectChanges();
   }
 
   buildMap() {
-    this.router.navigate(['../build-map'], { relativeTo: this.route });
+    this.creatingMap = true;
+    this.dialogRef = this.dialog.open(this.newMapDialog);
+  }
+
+  async mapDeleted() {
+    await this.getMaps();
+
+    this.selected = undefined;
+    this.goToMap();
   }
 
-  getMaps() {
-    let viewId: string;
-    this.route.params.subscribe(params => {
-      viewId = params['viewId']
-      console.log(viewId);
-    })
-    this.vmsService.getViewMaps(viewId).subscribe(data => {
-      console.log(data);
-      this.VmMaps = data;
-    })
+  // Get the available maps in this view
+  async getMaps() {
```

Comment:
Let's get together on Tuesday and discuss these changes. With angular development we want to use Observables from the rxjs package. instead of the built in javascript async and await features. I think you may need to create an akita state for your mapping portions, which will help you manage the Observables.